compat/cpu.h

```diff
@@ -9,23 +9,36 @@
 #ifndef SQUID_COMPAT_CPU_H
 #define SQUID_COMPAT_CPU_H
 
-#if HAVE_ERRNO_H
-#include <errno.h> /* for ENOTSUP */
-#endif
 #if HAVE_SCHED_H
 #include <sched.h>
 #endif
 
-#if !HAVE_CPU_AFFINITY
-/* failing replacements to minimize the number of if-HAVE_CPU_AFFINITYs */
-#if !HAVE_CPU_SET_T
+#if !HAVE_SCHED_SETAFFINITY
+
+#if HAVE_STDBOOL_H
+#include <stdbool.h>
+#endif
+#if HAVE_SYS_PARAM_H
+#include <sys/param.h>
+#endif
+#if HAVE_SYS_CPUSET_H
+#include <sys/cpuset.h>
+#if HAVE_CPUSET_T
+typedef cpuset_t cpu_set_t;
+#endif
+#define HAVE_CPU_AFFINITY 1
+inline void
+CpuAnd(cpu_set_t *destset, const cpu_set_t *srcset1, const cpu_set_t *srcset2)
+{
+    cpuset_t desttmp;
+    CPU_COPY(srcset1, &desttmp);
+    CPU_AND(&desttmp, srcset2);
+    CPU_COPY(&desttmp, destset);
+}
+#else
 typedef struct {
     int bits;
 } cpu_set_t;
```

Comment:
AFAICT, this code is now enabled when HAVE_SCHED_SETAFFINITY is set/true. Does HAVE_SCHED_SETAFFINITY really imply that we should be defining cpu_set_t?! Or did I misparsed the #if/#else combinations above?

Reply:
No, this else is when !HAVE_SCHED_SETAFFINITY and !HAVE_SYS_CPUSET_H

Reply:
Ah, I see that `#if` now. The code formatting/comments could be improved to make this clearer, but your answer addresses my change request.